qiskit_nature/operators/second_quantization/spin_op.py

```diff
@@ -12,117 +12,479 @@
-            new_operatorlist.append(other)
-            return SpinOp(new_operatorlist)
-
-        if isinstance(other, SpinOp):
-            new_operatorlist = copy.deepcopy(self.operator_list)
-            for elem in other.operator_list:
-                new_operatorlist.append(elem)
-                # If the operators have the same label, simply add coefficients:
-                for operator in new_operatorlist[:-1]:
-                    if elem.label == operator.label:
-                        new_operatorlist.pop()
-                        sum_coeff = operator.coeff + elem.coeff
-                        operator._coeff = sum_coeff  # set the coeff of sum operator to sum_coeff
-                        # if the new coefficient is zero, remove the operator from the list
-                        if sum_coeff == 0:
-                            new_operatorlist.remove(operator)
-                        break
-            return SpinOp(new_operatorlist)
-
-        raise TypeError("Unsupported operand type(s) for +: 'SpinOp' and "
-                        "'{}'".format(type(other).__name__))
+        return Fraction(self._dim - 1, 2)
 
-    def compose(self, other):
-        raise NotImplementedError()
-
-    def mul(self, other):
-        """Overloads the multiplication operator `*` for self and other, where other is a
-        number-type.
+    @property
+    def x(self) -> np.ndarray:
+        """A np.ndarray storing the power i of (spin) X operators on the spin system.
+        I.e. [0, 4, 2] corresponds to X_2^0 \\otimes X_1^4 \\otimes X_0^2, where X_i acts on the
+        i-th spin system in the register.
         """
-        if isinstance(other, numbers.Number):
-            # Create copy of the SpinOp in which every SpinOperator is multiplied by
-            # `other`.
-            new_operatorlist = [copy.deepcopy(base_operator) * other
-                                for base_operator in self.operator_list]
-            return SpinOp(new_operatorlist)
+        return self._spin_array[0]
 
-        raise TypeError("Unsupported operand type(s) for *: 'SpinOp' and "
-                        "'{}'".format(type(other).__name__))
+    @property
+    def y(self) -> np.ndarray:
+        """A np.ndarray storing the power i of (spin) Y operators on the spin system.
+        I.e. [0, 4, 2] corresponds to Y_2^0 \\otimes Y_1^4 \\otimes Y_0^2, where Y_i acts on the
+        i-th spin system in the register.
+        """
+        return self._spin_array[1]
 
     @property
-    def operator_list(self):
-        """operator list"""
-        return self._operator_list
+    def z(self) -> np.ndarray:
+        """A np.ndarray storing the power i of (spin) Z operators on the spin system.
+        I.e. [0, 4, 2] corresponds to Z_2^0 \\otimes Z_1^4 \\otimes Z_0^2, where Z_i acts on the
+        i-th spin system in the register.
+        """
+        return self._spin_array[2]
+
+    def __repr__(self) -> str:
```

Comment:
Do we have a convention in the project for the order of methods in a class? Usually, it might make sense to put magic methods right after the __init__ magic method.

Reply:
I am not sure for Qiskit, but you are right. I will change it.